Makefile

```diff
@@ -2,54 +2,177 @@
 #
 # Use of this source code is governed by the LICENSE file in this repository.
 
-build: binary-build
-
-run: build docker-run kubernetes-run
-
-docker-run: build docker-run
-
-kubernetes-run: build kubernetes-run
-
-#################################
-######      Go clean       ######
-#################################
+# The `clean` target is intended to clean the workspace
+# and prepare the local changes for submission.
+#
+# Usage: `make clean`
+.PHONY: clean
+clean: tidy vet fmt fix
 
-clean:
+# The `run` target is intended to build
+# and execute the binary.
+#
+# Usage: `make run`
+.PHONY: run
+run: build execute
 
+# The `tidy` target is intended to clean up
+# the Go module files (go.mod & go.sum).
+#
+# Usage: `make tidy`
+.PHONY: tidy
+tidy:
+	@echo
+	@echo "### Tidying Go module"
 	@go mod tidy
-	@go vet ./...
-	@go fmt ./...
-	@echo "I'm kind of the only name in clean energy right now"
 
-#################################
-######    Build Binary     ######
-#################################
+# The `vet` target is intended to inspect the
+# Go source code for potential issues.
+#
+# Usage: `make vet`
+.PHONY: vet
+vet:
+	@echo
+	@echo "### Vetting Go code"
+	@go vet ./...
 
-binary-build:
+# The `fmt` target is intended to format the
+# Go source code to meet the language standards.
+#
+# Usage: `make fmt`
+.PHONY: fmt
+fmt:
+	@echo
+	@echo "### Formatting Go Code"
+	@go fmt ./...
 
-	GOOS=darwin CGO_ENABLED=0 \
-		go build \
-		-o release/vela-runtime \
+# The `fix` target is intended to rewrite the
+# Go source code using old APIs.
+#
+# Usage: `make fix`
+.PHONY: fix
+fix:
+	@echo
+	@echo "### Fixing Go Code"
+	@go fix ./...
+
+# The `test` target is intended to run
+# the tests for the Go source code.
+#
+# Usage: `make test`
+.PHONY: test
+test:
+	@echo
+	@echo "### Testing Go Code"
+	@go test -race ./...
+
+# The `test-cover` target is intended to run
+# the tests for the Go source code and then
+# open the test coverage report.
+#
+# Usage: `make test-cover`
+.PHONY: test-cover
+test-cover:
```

Comment:
This `make` target is pretty nice because it will open up the test coverage report in the preferred browser.

It makes it easier to track down what LOC is and isn't covered by the tests.